### PR TITLE
Prevent power outages from randomly blinding AIs forever.

### LINKED ISSUE #11649: AI eye fuckery
(WEB REPORT BY: kavlax REMOTE: 172.93.109.202:7777)

> Revision da71ac1067b385e791746186ee75d9c1165848a3 
> 
> General description of the issue
> If the main breaker of an AI's APC is toggled off, the blind overlay will never go away even if the breaker is later toggled on.
> 
> What you expected to happen
> Wake me up
> 
> What actually happened
> Can't wake up
> 
> Steps to reproduce if possible
> Toggle breaker on your APC and weep.

### PR DESCRIPTION
**Pull request by Allspice**
Fixes #11649

### CODE REVIEW

**Comment by clusterfack** [comment]
It calls return right after unblindme = 1 :joy:

File: code/modules/mob/living/silicon/ai/life.dm [hunk 1]
```diff
@@ -168,8 +166,6 @@
 									show_laws()
 							sleep(50)
```
Comment (clusterfack) [inline code/modules/mob/living/silicon/ai/life.dm]: A sleep in the middle of Life() 😂 🔫

**Comment by clusterfack** [comment]
Just end it 😂 🔪 🖐

**Comment by Skullyton** [comment]
This isn't something that can be fixed simply by going through a single mob Life(), as it has been a problem with a lot of mobs regarding overlays (Both pain and blindness) in the fact that they seem to persist a lot longer than they should.

This should be approached a different way entirely, sorry to say.

**Comment by BarneyGumball** [comment]
@Skullyton These issues are caused by unrelated bits of bad code that don't set/unset overlays properly in certain instances.  Even using using overlay flags like some human overlays do has resulted in broken overlays because people don't use them properly.

**Comment by Skullyton** [comment]
@BarneyGumball Then by all means, divulge how they should be used properly for further reference. Something changed with overlays that changed their functionality entirely

File: code/modules/mob/living/silicon/ai/life.dm (hunk 1 above)
Comment (clusterfack) [inline code/modules/mob/living/silicon/ai/life.dm]: By the way in the case that this occurs it delays the entire mob dowork that goes through all the lifes

**Comment by BarneyGumball** [comment]
It's individual changes breaking them separately, no one commit is at fault for the overlay issues.  Some examples of oversights related to overlays:

This AI overlay bug we're talking about right now was an oversight where the proc was being `return`ed before the overlay could be unset: https://github.com/d3athrow/vgstation13/commit/b92ff00092acf24481502e92c11ad78ed489edfe

Welding goggles would flash their overlay because it would clear the 'tint' overlay before applying it again in `handle_regular_hud_updates.dm`: https://github.com/d3athrow/vgstation13/commit/31ac9f6a5bc78da1c7675f9d2166a62679f33866

Having damaged (but not broken) eyes would stop any blindness overlays because `eyes.dm` kept setting blindness to 0: https://github.com/d3athrow/vgstation13/commit/266628a9bcce35c2f0770ce52eb2cae7381519ac

(kek these are all sood's commit, this is not a bully post I swear)

File: code/modules/mob/living/silicon/ai/life.dm (hunk 1 above)
Comment (clusterfack) [inline code/modules/mob/living/silicon/ai/life.dm]: But yeah no this is a serious concern, change this to a spawn() at least

**Comment by gbasood** [comment]
AI lifecode will always be a mess

**Pull request merged**